test/helpers/writter.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helpers
+
+import (
+	"bytes"
+	"io"
+)
+
+// A Writer is a struct that has a variable-sized `bytes.Buffer` and one
+// outWriter(`io.writer`) to stream data
+type Writer struct {
+	buffer    *bytes.Buffer
+	outWriter io.Writer
+}
+
+// NewWriter creates and initializes a Writer with a empty Buffer and the given
+// outWriter
+func NewWriter(outWriter io.Writer) *Writer {
+	return &Writer{
+		buffer:    &bytes.Buffer{},
+		outWriter: outWriter,
+	}
+}
+
+// Write appends the contents of b to the buffer and outWriter, growing the
+// buffer as needed. The return value n is the length of p; err is always nil.
+// If the buffer becomes too large, Write will panic with ErrTooLarge.
+func (w *Writer) Write(b []byte) (n int, err error) {
```

Comment:
Seems like you implemented a https://golang.org/pkg/io/#MultiWriter here, please consider using it. This would entail calling `Reset` and `Bytes` on the buffer directly, but on the other hand they are only affecting the buffer, not all objects, so it seems like logical thing to do.